src/main/java/org/jitsi/jicofo/recording/jibri/JibriSession.java

```diff
@@ -335,127 +313,41 @@ public IQ processJibriIqFromJibri(JibriIq iq)
-    {
-        boolean doRetry
-            = error == null // on the first time there will be no error
-                || error.getExtension(
-                        "retry",
-                        "http://jitsi.org/protocol/jibri")
-                    != null;
-
-        logger.debug(
-            "Do retry? " + doRetry
-                + " retries: " + retryAttempt + " limit: " + NUM_RETRIES
-                + " in " + this.roomName);
-
-        if (doRetry && retryAttempt++ < NUM_RETRIES)
-        {
-            final EntityFullJid newJibriJid = jibriDetector.selectJibri();
-
-            logger.debug(
-                "Selected JIBRI: " + newJibriJid + " in " + this.roomName);
-
-            if (newJibriJid != null)
-            {
-                startJibri(newJibriJid);
-                return;
-            }
-            else if (error == null)
+            setJibriStatus(status, iq.getFailureReason());
+            if (JibriIq.Status.OFF.equals(status))
             {
-                // Classify this failure as 'service not available'
-                error
-                    = XMPPError.getBuilder(
-                        XMPPError.Condition.service_unavailable).build();
+                cleanupSession();
             }
         }
-        if (error == null)
-        {
-            error
-                = XMPPError.from(
-                    XMPPError.Condition.internal_server_error,
-                    "Retry limit exceeded").build();
-        }
-        // No more retries, stop either with the error passed as an argument
-        // or with one defined here in this method, which will provide more
-        // details about the reason
-        // Stop packet processor, etc.
-        stop(error);
-    }
-
-    /**
-     * Methods clears {@link #currentJibriJid} which means we're no longer
-     * recording nor in contact with any Jibri instance.
-     * Refreshes recording status in the room based on Jibri availability.
-     *
-     * @param error if the recording stopped because of an error it should be
-     * passed as an argument here which will result in stopping with
-     * the {@link JibriIq.Status#FAILED} status passed to the application,
-     * otherwise state will be {@link JibriIq.Status#OFF}.
-     */
-    private void recordingStopped(XMPPError error)
-    {
-        if (isSIP)
-        {
-            logger.info(
-                "Jibri SIP stopped for: "
-                    + sipAddress + " in: " + roomName);
-        }
         else
         {
-            logger.info("Recording stopped for: " + roomName);
+            logger.error("Received UNDEFINED status from jibri: " + iq.toString());
         }
 
-        currentJibriJid = null;
-        retryAttempt = 0;
+        return IQ.createResultIQ(iq);
+    }
 
-        // First we'll send an error and then follow with availability status
-        if (error != null)
+    public JibriIq.RecordingMode getRecordingMode()
```

Comment:
Since the method is public we need it to be documented.

Reply:
fixed